pkg/component/etcd/configmap/configmap_test.go

```diff
@@ -0,0 +1,255 @@
+		autoCompactionRetention string
+		labels                  map[string]string
+
+		cm *corev1.ConfigMap
+
+		values            *Values
+		configmapDeployer component.Deployer
+	)
+
+	BeforeEach(func() {
+		ctx = context.Background()
+		cl = fakeclient.NewClientBuilder().WithScheme(kubernetes.Scheme).Build()
+
+		name = "configmap"
+		namespace = "default"
+		uid = "a9b8c7d6e5f4"
+
+		metricsLevel = druidv1alpha1.Basic
+		quota = resource.MustParse("8Gi")
+		clientPort = 2222
+		serverPort = 3333
+		autoCompactionMode = "periodic"
+		autoCompactionRetention = "30m"
+
+		labels = map[string]string{
+			"foo": "bar",
+		}
+
+		etcd = &druidv1alpha1.Etcd{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: namespace,
+				UID:       uid,
+			},
+			Spec: druidv1alpha1.EtcdSpec{
+				Selector: metav1.SetAsLabelSelector(labels),
+				Replicas: 3,
+				Etcd: druidv1alpha1.EtcdConfig{
+					Quota:   &quota,
+					Metrics: &metricsLevel,
+					ClientUrlTLS: &druidv1alpha1.TLSConfig{
+						ClientTLSSecretRef: corev1.SecretReference{
+							Name: "client-url-etcd-client-tls",
+						},
+						ServerTLSSecretRef: corev1.SecretReference{
+							Name: "client-url-etcd-server-cert",
+						},
+						TLSCASecretRef: corev1.SecretReference{
+							Name: "client-url-ca-etcd",
+						},
+					},
+					PeerUrlTLS: &druidv1alpha1.TLSConfig{
+						TLSCASecretRef: corev1.SecretReference{
+							Name: "peer-url-ca-etcd",
+						},
+						ServerTLSSecretRef: corev1.SecretReference{
+							Name: "peer-url-etcd-server-tls",
+						},
+					},
+					ClientPort: &clientPort,
+					ServerPort: &serverPort,
+				},
+				Common: druidv1alpha1.SharedConfig{
+					AutoCompactionMode:      &autoCompactionMode,
+					AutoCompactionRetention: &autoCompactionRetention,
+				},
+			},
+		}
+
+		values = GenerateValues(etcd)
+
+		cm = &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("etcd-bootstrap-%s", string(values.EtcdUID[:6])),
+				Namespace: values.EtcdName,
+			},
+		}
+
+		configmapDeployer = New(cl, namespace, values)
+	})
+
+	Describe("#Deploy", func() {
+		Context("when configmap does not exist", func() {
+			It("should create the configmap successfully", func() {
+				Expect(configmapDeployer.Deploy(ctx)).To(Succeed())
+
+				cm := &corev1.ConfigMap{}
+
+				Expect(cl.Get(ctx, kutil.Key(namespace, values.ConfigMapName), cm)).To(Succeed())
+				checkConfigmap(cm, values)
+
+			})
+		})
+
+		Context("when configmap exists", func() {
+			It("should update the configmap successfully", func() {
+				Expect(cl.Create(ctx, cm)).To(Succeed())
+
+				Expect(configmapDeployer.Deploy(ctx)).To(Succeed())
+
+				cm := &corev1.ConfigMap{}
+
+				Expect(cl.Get(ctx, kutil.Key(namespace, values.ConfigMapName), cm)).To(Succeed())
+				checkConfigmap(cm, values)
+			})
+		})
+	})
+
+	Describe("#Destroy", func() {
+		Context("when services do not exist", func() {
+			It("should destroy successfully", func() {
```

Comment:
I don't understand the contexts of the tests in the `#Destroy` section
How is service in the picture?

Reply:
Mistake from my side